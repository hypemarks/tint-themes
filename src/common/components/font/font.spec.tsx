import { render } from '@testing-library/react';
import React from 'react';
import Font from './font';

describe('Test <Font /> Component', () => {
  it('should render snapshot', () => {
    const fontConfig = {
      fontFamily: 'fontFamily',
    };
    const wrapper = render(<Font fontConfig={fontConfig} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
  it('should render link html tag', () => {
    const fontConfig = {
      fontFamily: 'fontFamily',
    };
    const wrapper = render(<Font fontConfig={fontConfig} />);
    const link = wrapper.queryByTestId('font-link-tag');
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute(
      'href',
      `https://fonts.googleapis.com/css2?family=${fontConfig.fontFamily}:wght@300;400;700&display=swap" rel="stylesheet"`
    );
  });
});
