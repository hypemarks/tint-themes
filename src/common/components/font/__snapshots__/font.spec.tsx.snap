// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Font /> Component should render snapshot 1`] = `
<DocumentFragment>
  <link
    data-testid="font-link-tag"
    href="https://fonts.googleapis.com/css2?family=fontFamily:wght@300;400;700&display=swap\\" rel=\\"stylesheet\\""
    media="all"
    rel="stylesheet"
  />
</DocumentFragment>
`;
